File: test10/equi_filter.py
#!/usr/bin/python3

import re
import sys

lines = []
for line in sys.stdin:
   lines.append(line)

for line in lines:
   line = re.sub(r'\n$', '', line)
   words = re.split(r'\s+', line)

   for word in words:
      counter = {}
      for char in word:
         char = char.lower() 
         
         if char in counter:
            counter[char] += 1
         else:
            counter[char] = 1

      times = -1
      equi = True

      for char in counter.keys():
         if times == -1:
            times = counter[char]
         else:
            if times != counter[char]:
               equi = False
               break
      
      if equi:
         sys.stdout.write(word)
         sys.stdout.write(' ')

   sys.stdout.write('\n')

File: test10/equi_filter.pl
#!/usr/bin/perl -w

@lines = ();
while ($line = <STDIN>) {
   push @lines, $line;
}

foreach $line (@lines) {
   chomp $line;
   @words = split(/\s+/, $line);

   foreach $word (@words) {
      undef %counter;
      @characters = split(//, $word);

      foreach $char (@characters) {
         $char = lc $char;
         $counter{$char} += 1; 
      }

      $times = -1;
      $equi = 1;
      foreach $char (keys %counter) {
         if ($times == -1) {
            $times = $counter{$char};
         } else {
            if ($counter{$char} != $times) {
               $equi = 0;
               last;
            }
         }
      }

      print "$word " if ($equi);
   }
 
   print "\n";
}
